statistics/handle/lru_cache.go

```diff
@@ -173,6 +179,32 @@ func (s *statsInnerCache) updateIndices(tblID int64, tbl *statistics.Table, tblM
 	}
 }
 
+func (s *statsInnerCache) updateColumns(tblID int64, tbl *statistics.Table, tblMemUsage *statistics.TableMemoryUsage, needMove bool) {
+	element, exist := s.elements[tblID]
+	if exist {
+		oldtbl := element.tbl
+		deletedCol := make([]int64, 0)
+		for oldColID := range oldtbl.Columns {
+			_, exist := tbl.Columns[oldColID]
+			if !exist {
+				deletedCol = append(deletedCol, oldColID)
+			}
+		}
+		for colID, col := range tbl.Columns {
+			colMem := tblMemUsage.ColumnsMemUsage[colID]
+			s.lru.put(tblID, colID, false, col, colMem, true, needMove)
+		}
```

Comment:
This for-loop is duplicated with that out of "if exist", what about removing "return" and making this for-loop happen out of if-clause. So do "updateIndices".

Reply:
updated.